server/chunker.js

```diff
@@ -46,6 +57,24 @@ const getSemanticChunks = (text, chunkSize) => {
   const breakBySentences = (str) => breakByRegex(str, /(?<=[.。؟！?!\n])\s+/, true);
   const breakByWords = (str) => breakByRegex(str, /(\s,;:.+)/);
 
+  const breakByHtmlElements = (str) => {
+    const $ = cheerio.load(str, null, true);
+
+    // the .filter() call is important to get the text nodes
+    // https://stackoverflow.com/questions/54878673/cheerio-get-normal-text-nodes
+    let rootNodes = $('body').contents();
+
+    // create an array with the outerHTML of each node
+    const nodes = rootNodes.map((i, el) => $(el).prop('outerHTML') || $(el).text()).get();
+
+    // remove newlines from each node
+    nodes.forEach((node, i) => {
+        nodes[i] = node.replace(/\r?\n|\r/g, " ").trim();
+    });
+
+    return nodes.map(n => n + '\n\n').filter(n => n);
```

Comment:
I was getting this ready for merge (merged in main, etc.) and I noticed your additional tests in chunkfunction.test.js were failing because they didn't have the new 'html' input format specified.  So I started patching those up and ran into another error which is that compositing the nodes with '\n\n' is changing them from what you give as input and breaking the tests in several cases (no \n\n in input).  Looking at the code here, what you essentially do is strip any newlines out and then immediately add in two newlines when putting them back together.  This is lossy and in degenerative cases introduces newlines where none existed.  I propose just getting rid of both the strip and insert.  That makes the tests pass, but will it mess up any use cases you can think of?